kind/deploy.sh

```diff
@@ -0,0 +1,753 @@
+MAGENTA='\033[0;35m'
+RED='\033[0;31m'
+PURPLE="\033[0;35m"
+WHITE='\033[0;37m'
+YELLOW='\033[0;33m'
+
+log_message() {
+    VERBOSITY_LEVEL=$1
+    MESSAGE="${@:2}"
+    if [ "${LOGGING_VERBOSITY}" -ge "${VERBOSITY_LEVEL}" ]; then
+        echo -e "${MESSAGE}"
+    fi
+}
+
+repeat_char(){
+    COLOR=${1}
+	for i in {1..70}; do echo -ne "${!COLOR}$2${NC}"; done
+}
+
+msg() {
+    VERBOSITY_LEVEL=${1}
+    COLOR=${2}
+    MSG="${@:3}"
+    # echo -e "\n${!COLOR}## ${MSG}${NC}"
+    log_message ${VERBOSITY_LEVEL} "\n${!COLOR}## ${MSG}${NC}"
+}
+
+succeeded() {
+    VERBOSITY_LEVEL=$1
+    MSG="${@:2}"
+#   echo -e "${GREEN}NOTE:${NC} $1"
+    log_message ${VERBOSITY_LEVEL} "${GREEN}SUCCESS:${NC} ${MSG}"
+}
+
+note() {
+    VERBOSITY_LEVEL=$1
+    MSG="${@:2}"
+#   echo -e "${BLUE}NOTE:${NC} $1"
+    log_message ${VERBOSITY_LEVEL} "${BLUE}NOTE:${NC} ${MSG}"
+}
+
+warn() {
+#   echo -e "${YELLOW}WARN:${NC} $1"
+    log_message 1 "${YELLOW}WARN:${NC} $1"
+}
+
+error() {
+#   echo -e "${RED}ERROR:${NC} $1"
+    log_message 0 "${RED}ERROR:${NC} $1"
+}
+
+log() {
+    VERBOSITY_LEVEL=${1}
+    COLOR=${2}
+    MSG="${@:3}"
+    echo; repeat_char ${COLOR} '#'; msg ${1} ${MSG}; repeat_char ${COLOR} '#'; echo
+}
+
+#######################
+# /Logging and Output #
+#######################
+
+
+print_logo() {
+    log_message "1" ""
+    log_message "1" "Welcome to our"
+    log_message "1" "                                                                   "
+    log_message "1" "   _____                                  _                        "
+    log_message "1" "  / ____|                                | |                       "
+    log_message "1" " | (___    _ __     ___   __      __   __| |  _ __    ___    _ __  "
+    log_message "1" "  \___ \  | '_ \   / _ \  \ \ /\ / /  / _  | |  __|  / _ \  | \ _ \ "
+    log_message "1" "  ____) | | | | | | (_) |  \ V  V /  | (_| | | |    | (_) | | |_) |"
+    log_message "1" " |_____/  |_| |_|  \___/    \_/\_/    \__,_| |_|     \___/  |  __/ "
+    log_message "1" "                                                            | |    "
+    log_message "1" "                                                            |_|    "
+    log_message "1" " Kind installation script"
+    log_message "1" ""
+    log_message "1" "- Deploying a local secured (using htpasswd) docker registry"
+    log_message "1" "- Generating a selfsigned certificate (using openssl) to expose the registry as a HTTP/HTTPS endpoint"
+    log_message "1" "- Setting a docker network between the containers: kind and registry and alias \"registry.local\""
+    log_message "1" "- Allowing to access the repository using as address \"registry.local:${REGISTRY_PORT}\" within a pod, from laptop or when a pod is created"
+    log_message "1" "- Exposing 2 additional NodePort: 30000 and 31000"
+    log_message "1" "- Deploying an ingress controller"
+    log_message "1" "- Copying the generated certificate here: $HOME/local-registry.crt"
```

Comment:
As this text will depend about what the user will choose, I propose to remove it and instead, according to the parameters selected, that we show pat of the creation/installation what the script will currently do

Reply:
This text has been removed.